enhancements/baremetal/baremetal-provisioning-config.md

```diff
@@ -0,0 +1,169 @@
+the provisioning service to PXE boot a baremetal server and make it available as a
+BareMetal host. The configurations in this baremetal provisioning CR, are seperate
+from the configuration in the BareMetalHost CR. This new CR will allow for setting
+some sane defaults for these configurations with the ability to overrride them if 
+required. After the baremetal server is provisioned, the configurations in the
+BareMetalHost CR contain information to connect to the controller and also to
+fulfill a Machine.
+
+### Goals
+
+The goal of this enhancement request is to provide details about the configuration
+being added to a new CR used only in "metal3" deployments. This new CR would be
+within the operator scope and will only contain configuration required by the
+provisioning service to boot baremetal servers.
+
+### Non-Goals
+
+The provisioning network or the provisioning IP are not expected to change after
+deployment. But, it is possible for the DHCP range to be expanded after  few worker
+nodes have already been deployed. This proposal is not considering the update path
+for these configurations items.
+
+### Proposal
+
+Before BareMetal Hosts can be matched to Machines, they need to be connected to their
+provisioning network for them to be PXE booted and given an IP address. To make
+this happen, the provisioning service needs to know which NIC, provisioning network,
+IP address and image URLs to use to download and boot images on these servers.
+
+A new CR called "Metal3ProvisioningController" or just "Metal3Controller" would be
+created within the operator scope.
+
+This new CR would consist of the following:
+
+1. ProvisioningInterface : This is the interface name on the underlying baremetal
+server which would be physically connected to the provisioning network. This
+configuration is needed only for the underlying provisioning service (Ironic)
+and could have values like "eth1" or "ens3".
+
+2. ProvisioningIP : This is the IP address used to to bring up a NIC on the
+baremetal server for provisioning. This is also a value that is useful just to the
+provisioning service. This value should not be in the DHCP range and should not
+be used in the provisioning network for any other purpose (should be a free IP
+address.) It is expected to be provided in the format : 10.1.0.3.
+
+3. ProvisioningNetworkCIDR : This is the provisioning network with its CIDR. The
+ProvisioningIP and the ProvisioningDHCPRange are expected to be within this network.
+The value for this config is expected to be provided in the format : 10.1.0.0/24.
+
+4. ProvisioningDHCPRange : This is a range of IP addresses from which baremetal
+hosts can be assigned their IP address. This range should be provided as two
+comma seperated IP addresses where the addresses represent a contiguous range of
+IPs that are available to be allocated. It is expected to be provided in the format :
+10.1.0.10, 10.1.0.100.
```

Comment:
We could mention that this will be optional, and if the range specified is empty/null no DHCP services will be started, for environments where external DHCP is provided for the provisioning network?

Reply:
Yes. Will add that too.